### PR TITLE
Add appear animation to phppresentation for Powerpoint2007 writer

### PR DESCRIPTION
**Pull request by JewrassicPark**
Added animation class, added animation tags to Powerpoint2007 Slide Writer, added animation objects to Slide.php

This only supports appear. Shapes can be grouped to appear together or sequentially by order they're added as animations. 

Use:

```
$animation = new Animation(); //create the animation
$currentSlide = $this->pp->getActiveSlide();  //get the current slide
$currentSlide->addAnimation($animation); //add the animation to the current slide
/*
* add n shapes to animation
* every shape in an animation will appear at the same time on click
*/
$animation->addShape($shape); 
$animation->addShape($shape2); 
```

### CODE REVIEW

**Comment by Progi1984** [comment]
@JewrassicPark Before merging, you need to fix PHPCS & PHPMd errors :
https://travis-ci.org/PHPOffice/PHPPresentation/jobs/105507570

**Comment by JewrassicPark** [comment]
ive updated the style in my fork

File: src/PhpPresentation/Writer/PowerPoint2007/Slide.php [hunk 1]
```diff
@@ -180,7 +180,7 @@ public function writeSlide(SlideElement $pSlide = null)
         $objWriter->endElement();
 
         // Loop shapes
-        $shapeId = 0;
+        $shapeId = 1;
```
Comment (Progi1984) [inline src/PhpPresentation/Writer/PowerPoint2007/Slide.php]: Why have you changed this ?

File: src/PhpPresentation/Slide.php [hunk 2]
```diff
@@ -515,4 +521,27 @@ public function setBackground(AbstractBackground $background = null)
...
+    /**
+     * Get collection of animations
+     *
+     * @return \PhpOffice\PhpPresentation\Slide\Animation
+     */
+    
+    public function getAnimations()
+    {
+        return $this->animations;
+    }
```
Comment (Progi1984) [inline src/PhpPresentation/Slide.php]: That lacks a setAnimations(array $animations = array())

**Comment by Progi1984** [comment]
@JewrassicPark Could you add a sample ? Which type of animation could you define ?

**Comment by JewrassicPark** [comment]
It only does appear animation currently

Here is a use example: 

```
<?php


	$root = "/php/PHPPresentation/PHPPresentation-develop-Jan-2016";
		
	use PhpOffice\PhpPresentation\PhpPresentation;
	use PhpOffice\PhpPresentation\Shape\RichText;
	use PhpOffice\PhpPresentation\IOFactory;
	use PhpOffice\PhpPresentation\Style\Fill;
	use PhpOffice\PhpPresentation\Slide\Animation;

	require_once $root.'/src/PhpPresentation/Autoloader.php';
	\PhpOffice\PhpPresentation\Autoloader::register();

	use PhpOffice\Common;
	$commonRoot = "/php/PHPOffice-Common-7-29-2015";
	require_once $commonRoot.'/src/Common/Autoloader.php';
	\PhpOffice\Common\Autoloader::register(); 
	
	//Create Presentation
	$pp = new PhpPresentation();
	$pp->getProperties()->setCreator('Test')
			  ->setLastModifiedBy('Test')
			  ->setTitle('Test Animation PPT')
			  ->setSubject('Test Animation Example')
			  ->setDescription('Testing the appear animation')
			  ->setKeywords('powerpoint animation test')
			  ->setCategory('Testing');
	
	//Start Example
	$currentSlide = $pp->getActiveSlide(); 
	
	//add a shape to apply the animation to
	$shape = $currentSlide->createRichTextShape()->setWidth(100)->setHeight(100)->setOffsetX(100)->setOffsetY(100)->setAutoFit(RichText::AUTOFIT_NOAUTOFIT);
	$shape->getFill()->setFillType(Fill::FILL_SOLID);
	$shape->getFill()->getStartColor()->setRGB('FF0000');
	
	//add a shape to demonstrate shape grouping	
	$shape2 = $currentSlide->createRichTextShape()->setWidth(100)->setHeight(100)->setOffsetX(300)->setOffsetY(300)->setAutoFit(RichText::AUTOFIT_NOAUTOFIT);
	$shape2->getFill()->setFillType(Fill::FILL_SOLID);
	$shape2->getFill()->getStartColor()->setRGB('FF0000');
	
	
	//create animation object
	$animation = new Animation();
	//add the shape we made to the animtion
	$animation->addShape($shape);
	//add the second shape 
	$animation->addShape($shape2); 
	
	
	//add the animation to the slide
	$currentSlide->addAnimation($animation);
	
	//create another shape to demonstrate animation order 
	$shape3 = $currentSlide->createRichTextShape()->setWidth(100)->setHeight(100)->setOffsetX(400)->setOffsetY(400)->setAutoFit(RichText::AUTOFIT_NOAUTOFIT);
	$shape3->getFill()->setFillType(Fill::FILL_SOLID);
	$shape3->getFill()->getStartColor()->setRGB('00FF00');

	//make second animation 
	$animation2 = new Animation(); 
	//add the new shape
	$animation2->addShape($shape3);
	//add the second animation to the slid
	$currentSlide->addAnimation($animation2); 

	//save powerpoint
	$fullPath = "test-ppt.pptx";
	@unlink($fullPath);
	$oWriterPPTX = IOFactory::createWriter($pp, 'PowerPoint2007');
	$oWriterPPTX->save($fullPath);		
?>
```

**Comment by Progi1984** [comment]
I'll work with #179 before this one :)

File: src/PhpPresentation/Writer/PowerPoint2007/Slide.php (hunk 1 above)
Comment (Progi1984) [inline src/PhpPresentation/Writer/PowerPoint2007/Slide.php]: @JewrassicPark Why have you changed this ?

**Comment by JewrassicPark** [comment]
It seemed like powerpoint was treating shape 0 as the slide itself. IIRC I couldn't get the animations to work when the shape index started at 0.

**Comment by Progi1984** [comment]
@JewrassicPark Animations or Slide notes ?

**Comment by JewrassicPark** [comment]
I believe that was changed specifically for animations. I'm not sure if it affects the slide notes either way.

**Comment by Progi1984** [comment]
@JewrassicPark Could you update your PR with the latest version of the develop branch ?

**Comment by JewrassicPark** [comment]
I made a new pull request with a fork from the latest branch. I deleted my old fork, forked from the latest dev branch, added in my changes for animation and tested it. Is there a better way to do that? It was a pain in the ass and I'm not going to be doing it often.

**Pull request closed without merging**